### ISSUE TITLE
Multi-GPU Training freezes

### ISSUE THREAD

**Issue by arthitag** [pytorch/pytorch#18879]
Training abruptly and unpredictably stops after several iterations. There’s no error message. The process hangs.

Environment

PyTorch Version (e.g., 1.0): 0.3.1
OS (e.g., Linux): RHEL7.6
How you installed PyTorch (conda, pip, source): virtual environment, pip
Build command you used (if compiling from source):
Python version: 2.7.5
CUDA/cuDNN version: 8.0.44, 6.0
GPU models and configuration: 4 NVIDIA P6000s
Any other relevant information: There were no issues with training the same model on single GPU
Similar code with another slightly lighter model (with fewer outputs for loss computation) trains without any issues.
Things I have tried to resolve the issue but didn’t work

Reduce number of dataloader threads from 4 to 1
Reducing batch size from 24 to 12
Reducing model size using point-wise convolution
Computing loss in forward function

**Comment by fmassa** [comment]
You are using a pretty old PyTorch version, and many things have improved since 0.3.1 was released.
I'd recommend you update to latest PyTorch version and try again.

I'm closing this issue, but please let us know if your code still hangs with the latest version of PyTorch.
In this case, sharing the code / a snippet and providing more information of the specificities of the model would help a lot.
